### PR TITLE
Add RTTDelegate and code that calls it

### PR DESCRIPTION
**Pull request by derekchiang**

### CODE REVIEW

**Comment by armon** [comment]
@derekchiang Looks good, but we need some tests ;)

**Comment by derekchiang** [comment]
Ahhhh I hate tests

**Comment by mitchellh** [comment]
Tests are an investment, you love them when they save you from walking into a snake trap. :)

**Comment by derekchiang** [comment]
@armon

File: memberlist_test.go [hunk 1]
```diff
@@ -1046,6 +1046,53 @@ func TestMemberlist_conflictDelegate(t *testing.T) {
...
+}
+
+func (m *MockRTT) NotifyRTT(other *Node, _ time.Duration) {
+	m.other = other
+}
+
+func TestMemberlist_RTTDelegate(t *testing.T) {
+	m1 := GetMemberlist(t)
+	m1.setAlive()
+	m1.schedule()
```
Comment (armon) [inline memberlist_test.go]: You should defer m1.Shutdown()

File: memberlist_test.go [hunk 2]
```diff
@@ -1046,6 +1046,53 @@ func TestMemberlist_conflictDelegate(t *testing.T) {
 	}
 }
 
+type MockRTT struct {
+	other *Node
```
Comment (armon) [inline memberlist_test.go]: Let's store the RTT

File: memberlist_test.go [hunk 3]
```diff
@@ -1046,6 +1046,53 @@ func TestMemberlist_conflictDelegate(t *testing.T) {
...
+
+	// Ensure we were notified
+	if mock.other == nil {
+		t.Fatalf("should get notified")
+	}
+
+	if !reflect.DeepEqual(mock.other, m1.LocalNode()) {
+		t.Fatalf("not notified about the correct node; expected: %+v; actual: %+v",
+			m2.LocalNode(), mock.other)
+	}
```
Comment (armon) [inline memberlist_test.go]: Also assert RTT > 0

**Comment by armon** [comment]
@derekchiang Left some minor feedback

**Comment by derekchiang** [comment]
@armon feedback addressed.

**Comment by armon** [comment]
LGTM

**Pull request merged**